Documentation/ApiOverview/MessageBus/Index.rst

```diff
@@ -0,0 +1,238 @@
+..  include:: /Includes.rst.txt
+..  index:: Message Bus
+..  _message-bus:
+
+===========
+Message bus
+===========
+
+..  versionadded:: 12.2
+
+TYPO3 provides a message bus solution based on `symfony/messenger`_. It has the
+ability to send messages and then handle them immediately (synchronous) or
+send them through transports (asynchronous, for example, queues) to be handled
+later.
+
+For backwards compatibility, the default implementation uses the synchronous
+transport. This means that the message bus will behave exactly as before, but it
+will be possible to switch to a different (asynchronous) transport on a
+per-project base.
+
+To offer asynchronicity, TYPO3 also provides a transport implementation based on
+the `Doctrine DBAL messenger transport`_ from Symfony and a basic implementation
+of a consumer command.
+
+
+..  contents:: Table of Contents
+    :local:
+
+
+"Everyday" usage - as a developer
+=================================
+
+Dispatch a message
+------------------
+
+..  rst-class:: bignums
+
+#.  Add a PHP class for your message object (which is an arbitrary PHP class)
+
+    ..  literalinclude:: _DemoMessage.php
+        :caption: EXT:my_extension/Classes/Queue/Message/DemoMessage.php
+
+#.  Inject the :php:`MessageBusInterface` into your class and call the
+    :php:`dispatch()` method
+
+    ..  literalinclude:: _MyClass.php
+        :caption: EXT:my_extension/Classes/MyClass.php
+
+
+Register a handler
+------------------
+
+..  rst-class:: bignums
+
+#.  Implement the handler class
+
+    ..  literalinclude:: _DemoHandler.php
+        :caption: EXT:my_extension/Classes/Queue/Handler/DemoHandler.php
+
+#.  Register the handler
+
+    Use a tag to register a handler. Use :yaml:`before`/:yaml:`after` to define
+    an order.
+
+    ..  literalinclude:: _demo-handler.yaml
+        :language: yaml
+        :caption: EXT:my_extension/Configuration/Services.yaml
+
+"Everyday" usage - as a system administrator/integrator
+=======================================================
+
+By default, TYPO3 will behave like in versions before TYPO3 v12. This means that
+the message bus will use the synchronous transport and all messages will be
+handled immediately. To benefit from the message bus, it is recommended to
+switch to an asynchronous transport. Using asynchronous transports increases the
+resilience of the system by decoupling external dependencies even further.
```

Comment:
why "even further", decoupling wasn't mentioned before

Reply:
A message bus decouples the code by using it. The class dispatches a message and thats it (for example for sending an email). The class itself does not handling the sending of the email, but the messenger component does. So the code it decoupled - also with the synchronous transport.